### DISCUSSION TITLE
Disable export of extra field?

### DISCUSSION THREAD

**Discussion by cdriveraus** [retorquere/zotero-better-bibtex#2662] [General]
The ability to disable export of certain fields is useful, but I face the problem that some references get imported to zotero with additional fields populated with things I don't want in my reference list at the end, but better-biblatex exports all of these, and the apa latex package includes them. It would be very useful to disable the extra field from being exported at all, but this doesn't seem to work -- is this because it's not a real field in the sense that the better bibtex option is referring to, but a collection of all possible fields? Any ideas to work around this?

**Comment by retorquere** [comment]
In general [this](https://retorque.re/zotero-better-bibtex/installation/preferences/export/#fields-to-omit-from-export-comma-separated) will allow you to specify output fields (such as `note`, or `annotation`) to remove from the output. If that doesn't do what you need, I'd need a debug log for the item you're looking at.

**Reply by cdriveraus** [reply]
Ok, it doesn't work with respect to the Extra field -- I specify that
```
extra,Extra
```
should not be included in exports, but information from Extra is included in the resulting .bib when I export using better biblatex
[HLM7FSNG-refs-euc.zip](https://github.com/retorquere/zotero-better-bibtex/files/12725621/HLM7FSNG-refs-euc.zip)

**Reply by retorquere** [reply]
That feature allows selecting _output_ fields for removal (and it's case insensitive). In this case your `extra` field has

```
Publisher: American Educational Research Association
Citation Key: pianta2008classroom
```

Citation key shows up as the citation key, `American Educational Research Association` shows up in `publisher`; if you want that information removed from the output you would need to enter `publisher` in the skipfield list. But that would remove `publisher` everywhere, also when it comes from non-`extra` fields.

There is no feature to skip input.

But that information in the `extra` field has meaning in Zotero -- it will also show up in Zotero-generated bibliographies. If you don't want `American Educational Research Association` to show up in the output it seems better to remove it from `extra` (maybe by moving it to a note item.

**Reply by cdriveraus** [reply]
Ok, thanks. It seems that the default zotero biblatex export *doesn't* include such extra details, and for whatever reason the extra field often gets populated by a number of things that are not desired when using the automatic import feature of zotero from various web services -- hence, this problem only shows up when using better bibtex, even though I agree that it's not doing anything obviously wrong or obnoxious and it's a problem I face because of either the apa latex package or the various web services providing information...  yes the solution so far has been to always remove manually, it's just annoying.

**Reply by retorquere** [reply]
I agree this can be a nuisance. The status of the `extra` field has never been formally specified to my knowledge, and there are loads of people using it as a notes field. But that's where Zotero stores and reads "extra field" information.

<details><summary>I have a tendency to lecture when talking about such things. None of what's below is essential to know</summary>

What is entered there is treated as meaningful info in the Zotero => Word/LibreOffice/GDocs pipeline, not just notes, and BBT has as a criteria-when-sensibly-possible to output bib(la)tex that would render to the same or as close as possible to what Zotero would render . And that pipeline will pick up the `Publisher`. The transformation to any format should preserve the meaning/intent of the input where possible, and the standard bib(la)tex exporters don't do that; I consider that a flaw.

</details>

**Reply by retorquere** [reply]
I maybe have a solution... can you open a new issue?

**Reply by retorquere** [reply]
But in this case it's the publisher field you want to get rid of? There's no remaining non-parsable text.

**Reply by cdriveraus** [reply]
that was just one example of many, not specific to the publisher field.

**Reply by retorquere** [reply]
See #2663 for further discussion.

**Comment by cdriveraus** [comment]
https://github.com/retorquere/zotero-better-bibtex/issues/2663

**Comment by cdriveraus** [comment]
> What is entered there is treated as meaningful info in the Zotero => Word/LibreOffice/GDocs pipeline, not just notes, and BBT has as a criteria-when-sensibly-possible to output bib(la)tex that would render to the same or as close as possible to what Zotero would render . And that pipeline will pick up the `Publisher`. The transformation to any format should preserve the meaning/intent of the input where possible, and the standard bib(la)tex exporters don't do that; I consider that a flaw.

Always a tension between doing what's idealistically 'best' and dealing with the real world right :)

**Reply by retorquere** [reply]
There's no optimal solution, but I aim to be consistent, there's the intent-preservation principle, and I don't take the standard bib(la)tex exporters as normative for anything. BBT is after all better.
